arbos/tx_processor.go

```diff
@@ -270,6 +276,13 @@ func (p *TxProcessor) GasChargingHook(gasRemaining *uint64) (*common.Address, er
 
 		// Pad the L1 cost by 10% in case the L1 gas price rises
 		posterCost = arbmath.BigMulByFrac(posterCost, 110, 100)
+	} else {
+		posterCost, calldataUnits, _ = p.state.L1PricingState().PosterDataCost(p.msg, p.evm.Context.Coinbase)
+		if calldataUnits > 0 {
+			if err := p.state.L1PricingState().AddToUnitsSinceUpdate(calldataUnits); err != nil {
```

Comment:
We should think about what we want the behavior for `eth_call`s to be. If we don't want them to include the poster data cost, as this PR changes it to, we need to update the geth code to increase the gas limit for calls by the data cost.

We should also consider having a variable for the poster that we assign to to deduplicate this code.

Reply:
Refactored to deduplicate code. `eth_call` behavior issue is still open for discussion.

Reply:
I believe in classic we include the poster data cost in `eth_call`
It keeps values consistent with the return of estimateGas
Also makes the call execution closer to what would be in an actual tx